shell/common/shorebird/shorebird.cc

```diff
@@ -79,6 +79,116 @@ FileCallbacks ShorebirdFileCallbacks() {
   };
 }
 
+std::string ConfigureShorebird(const ShorebirdConfigArgs& args) {
+  FML_LOG(INFO) << "In ConfigureShorebird";
+  // FIXME: This was commented out because the windows flutter engine does not
+  //     populate the settings snapshots. Ideally we would call
+  //     ConfigureShorebird from the embedder and this would be uncommented.
+  // If you are crashing here, you probably are running Shorebird in a Debug
+  // config, where the AOT snapshot won't be linked into the process, and thus
+  // lookups will fail.  Change your Scheme to Release to fix:
+  // https://github.com/flutter/flutter/wiki/Debugging-the-engine#debugging-ios-builds-with-xcode
+  // FML_CHECK(DartSnapshot::VMSnapshotFromSettings(settings))
+  //     << "XCode Scheme must be set to Release to use Shorebird";
+
+  auto shorebird_updater_dir_name = "shorebird_updater";
+
+  auto code_cache_dir = fml::paths::JoinPaths(
+      {std::move(args.code_cache_path), shorebird_updater_dir_name});
+  auto app_storage_dir = fml::paths::JoinPaths(
+      {std::move(args.app_storage_path), shorebird_updater_dir_name});
+
+  fml::CreateDirectory(fml::paths::GetCachesDirectory(),
+                       {shorebird_updater_dir_name},
+                       fml::FilePermission::kReadWrite);
+
+  bool init_result;
+  // Using a block to make AppParameters lifetime explicit.
+  {
+    FML_LOG(INFO) << "Constructing app parameters";
+    AppParameters app_parameters;
+    // Combine version and version_code into a single string.
+    // We could also pass these separately through to the updater if needed.
+    auto release_version = args.version + "+" + args.version_code;
+    app_parameters.release_version = release_version.c_str();
+    app_parameters.code_cache_dir = code_cache_dir.c_str();
+    app_parameters.app_storage_dir = app_storage_dir.c_str();
+
+    // https://stackoverflow.com/questions/26032039/convert-vectorstring-into-char-c
```

Comment:
I think we do this in the Android codepath too.

Reply:
Yep, this function and the other ConfigureShorebird need to be consolidated. Doing that as a followup PR, as it affects the macOS bits

Reply:
I meant specifically the "turn a vector<string> into vector<char*>